### ISSUE TITLE
[ENHANCEMENT] Provide CDI AddOn

### ISSUE THREAD

**Issue by mircea-pavel-anton** [harvester/harvester#4470]
**Is your enhancement related to a problem? Please describe.**

I want to deploy a bunch of identical virtual machines using a `VirtualMachinePool` object from kubevirt. In order to provide persistent storage to each VM, as far as i understand from the docs, I would need to install the CDI: https://kubevirt.io/user-guide/virtual_machines/pool/#using-persistent-storage

**Describe the solution you'd like**

Just like we have the pcie devices controller as an add-on, we could possibly have the CDI as an addon as well.

**Describe alternatives you've considered**

Alternatively, the CDI could be manually installed.

**Additional context**

Basically what I am trying to achieve is to deploy a Talos Linux cluster using GitOps. I would like to deploy FluxCD and then create virtual machine pools for the controlplane and worker nodes of the cluster so that I can scale them up and down.

**Comment by andyrue** [comment]
I just came here to request this feature too. In my case I'm wanting to create `VirtualMachines` with a data volume that is populated from a container registry image by using the `dataVolume` spec which is ultimately what's needed for a `VirtualMachinePool` too. https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#datavolume

I'll add to this, it would also be useful to be able to specify a `containerDisk` that would use a `dataVolume` from within the Container Volume GUI so when using `containerDisks`, they are no longer ephemeral. Could be something as simple as a checkbox with a hint as to what it does. `Use DataVolume ✅ - "Causes your container data to persist"` or something like that. Having this available through the Terraform Provider would also be a welcome addition.

**Comment by gibson3659** [comment]
I also need cdi but for uploading images from the command line.  Not all images are available on http servers, so what about windows shares and sftp.  I also shouldn't be forced use a remote desktop or to download a large image just so I can upload it using the web page, which has its own issues with large images.  There should always be a cli option that can use local files to fully support remote work.

My current workaround is a bit ridiculous, but it works.  I spin-up a small linux VM in harvester with volumes to hold my images, mount my remote samba share, use qemu-img to dd the image to the attached volumes.  Convert the volumes to a vm image.  "virtctl image-upload" support would be much more reasonable.
